lib/devise/models/database_authenticatable.rb

```diff
@@ -37,11 +37,12 @@ def self.required_fields(klass)
       # the hashed password.
       def password=(new_password)
         @password = new_password
-        self.encrypted_password = password_digest(@password) if @password.present?
+        self.encrypted_password = password_digest(@password) 
```

Comment:
By the looks of it we need to update `clean_up_passwords` to clear the instance variables directly, instead of calling `password=`.

```diff
       def clean_up_passwords
-        self.password = self.password_confirmation = nil
+        @password = @password_confirmation = nil
       end
```